src/tachimawari/control/controller/module/cm740.cpp

```diff
@@ -0,0 +1,319 @@
+#include "tachimawari/control/packet/protocol_1/instruction/instruction.hpp"
+#include "tachimawari/control/packet/protocol_1/instruction/sync_write_packet.hpp"
+#include "tachimawari/control/packet/protocol_1/instruction/write_packet.hpp"
+#include "tachimawari/control/packet/protocol_1/model/packet_id.hpp"
+#include "tachimawari/control/packet/protocol_1/status/bulk_read_data.hpp"
+#include "tachimawari/control/packet/protocol_1/status/status_packet.hpp"
+#include "tachimawari/control/packet/protocol_1/utils/word.hpp"
+#include "tachimawari/joint/model/joint.hpp"
+#include "tachimawari/joint/protocol_1/mx28_address.hpp"
+
+#include "errno.h"  // NOLINT
+#include "fcntl.h"  // NOLINT
+#include "stdio.h"  // NOLINT
+#include "string.h"  // NOLINT
+#include "termios.h"  // NOLINT
+#include "unistd.h"  // NOLINT
+#include "linux/serial.h"
+#include "sys/ioctl.h"
+#include "sys/time.h"
+
+namespace tachimawari::control
+{
+
+CM740::CM740(
+  const std::string & port_name, int baudrate,
+  float protocol_version)
+: ControlManager(port_name, protocol_version, baudrate), byte_transfer_time(0.0),
+  platform(std::make_shared<Linux>()),
+  bulk_data(std::make_shared<std::map<uint8_t, packet::protocol_1::BulkReadData>>())
+{
+}
+
+bool CM740::connect()
+{
+  if (platform->open_port(port_name, baudrate)) {
+    byte_transfer_time = (1000.0 / baudrate) * 12.0;
+
+    return dxl_power_on();
+  }
+
+  return false;
+}
+
+bool CM740::dxl_power_on()
+{
+  if (write_packet(CM740Address::DXL_POWER, 1)) {
+    if (protocol_version == 1.0) {
+      write_packet(CM740Address::LED_HEAD_L, packet::protocol_1::Word::make_color(255, 128, 0), 2);
+    }
+
+    return true;
+  }
+
+  return false;
+}
+
+packet::protocol_1::StatusPacket CM740::send_packet(packet::protocol_1::Packet packet)
+{
+  {
+    using packet::protocol_1::StatusPacket;
+
+    std::vector<uint8_t> txpacket = packet.get_packet();
+
+    if (platform->write_port(txpacket) == txpacket.size()) {
+      // set packet timeout;
+      int get_length = 0;
+      int expected_length = packet.get_expected_length();
+      auto rxpacket = std::make_shared<std::vector<uint8_t>>(MAX_PACKET_SIZE, 0x00);
+
+      while (true) {
+        get_length += platform->read_port(rxpacket, expected_length - get_length, get_length);
+
+        if (get_length == expected_length) {
+          int new_get_length = StatusPacket::validate(rxpacket, get_length);
+
+          if (new_get_length == get_length) {
+            return StatusPacket(rxpacket, get_length);
+          } else if (new_get_length != 0) {
+            // is packet timeout ? so RX_TIMEOUT
+            get_length = new_get_length;
+          } else {
+            // so RX_CORRUPT
+            return StatusPacket(nullptr, 0);
+          }
+        } else {
+          // is packet timeout ? so RX_TIMEOUT
+          if (get_length > expected_length) {
+            break;
+          }
+        }
+      }
+    } else {
+      // so TX_FAIL
```

Comment:
To be done? Could you check other lines too?

Reply:
Perhaps, I will put the `TODO` comment on it.